### PR TITLE
Conversion to Graph Database

### PR DESCRIPTION
**Pull request by DMalone87**
- Done: Convert to graph database
- Done: Update requirements
- Done: Remove SQLAlchemy
- Done: Update API to support db changes
- Done: Complete the data model conversion
- Done: Update tests to use the new DB
- Pass: Add database migration
    - This will be handled in a future update.

### CODE REVIEW

**Comment by DMalone87** [comment]
Most recent update finally has the Graph DB image replacing the PGSQL one. I'm using version [5.23-community](https://hub.docker.com/layers/library/neo4j/5.23-community/images/sha256-77918f6f6b94970d824304f74f38cade16592b52acc2fdf0b681fcd9d47776cd?context=explore)

**Comment by DMalone87** [comment]
With the most recent update, the API works for most endpoints. Core CRUD functions are there. I'll most likely move on to getting the testing and migration infrastructure settled and handle the rest of the API function in a future PR. There will be some big changes coming to the API to bring it in line with the new API spec, so it's best to handle that at all at once.

File: .github/workflows/frontend-checks.yml [hunk 1]
```diff
@@ -28,11 +28,3 @@ jobs:
...
         if: always()
         run: npm run check-formatting
-
-      - name: Jest Tests
-        if: always()
-        run: npm run test
-
-      - name: Types
-        if: always()
-        run: npm run check-types
```
Comment (zganger) [inline .github/workflows/frontend-checks.yml:38]: why remove frontend tests?

**Review by zganger — COMMENTED** [review]
A few code style failures in test- will prevent this from being deployed. Though without db migration that may not matter.

File: .github/workflows/frontend-checks.yml (hunk 1 above)
Comment (DMalone87) [reply .github/workflows/frontend-checks.yml:38]: The front end is going to change significantly pretty soon so I didn't think it was that big of a priority. 

That and I didn't really understand the test that was failing at the time. It was something to do with the registration page... The page runs and completes registration successfully though.

**Review by zganger — APPROVED** [review]

**Pull request merged**